avidemux_plugins/ADM_videoFilters6/asharp/qt4/Q_asharp.cpp

```diff
@@ -102,6 +105,17 @@ void Ui_asharpWindow::showEvent(QShowEvent *event)
 }
 
 #define MYSPIN(x) w->doubleSpinBox##x
+#define MYSLIDER(x) w->horizontalSlider##x
+
+void Ui_asharpWindow::valueChangedSlider(int f)
+{
+	Ui_asharpDialog *w=(Ui_asharpDialog *)myCrop->_cookie;
```

Comment:
We don't need to go this way as we stay within the class, do we? Is the purpose just to re-use the define?

Reply:
IMHO fever or none macro re-define is better solution.
Should i change it to `Ui_asharpDialog *w=&ui;` ?

Reply:
I meant e.g.

```
void Ui_asharpWindow::valueChangedSlider(int f)
{
#define SYNCSPIN(x) ui.doubleSpinBox##x->setValue(ui.horizontalSlider##x->value() / 100.);
    SYNCSPIN(Treshold)
    SYNCSPIN(Strength)
    SYNCSPIN(Block)

    valueChanged(0);
}
```
Explicit cast is no needed.

Reply:
My bad, explicit cast here needed, I got lost in brackets.

Reply:
I can rewrite if you want.

Reply:
I would suggest I pull it now and fix a few remaining UX issues (tab order, single step for spinboxes) myself.